src/main/java/org/commcare/cases/query/queryset/DerivedCaseQueryLookup.java

```diff
@@ -0,0 +1,77 @@
+package org.commcare.cases.query.queryset;
```

Comment:
This class and its inheritor also seem an unnecessary split that makes the code hard to read. Unless you have a very explicit reason for keeping its parent abstract class the logic seems so paired that they could be merged. I'm not clear what logic belongs in here and what in its inheritors like CaseIndexQueryTransform. It also doesn't look like any of the QuerySetLookup methods are called in the tests so I can't tell what's going on.